openedx/core/djangoapps/content_libraries/docs/decisions/0002-libraries-lti-provider.rst

```diff
@@ -0,0 +1,30 @@
+Allow content libraries to used by LTI consumers
+------------------------------------------------
+
+Status
+------
+
+Pending
+
+Context
+-------
+
+Currently, there's no way for xblocks in blockstore-based content libraries to be served to other LTI consumers. This ADR explores ways in which this can be achieved, without overly complicating the codebase. The platform currently does have an :code:`lms.lti_provider` app which implements the LTI 1.1 spec, but it only works for modulestore-based courses, and the LTI 1.1 spec itself is also deprecated and deemed insecure.
+
+Decision
+--------
+
+We will use the latest LTI spec (`LTI 1.3`_) to serve the xblocks. This would be largely independent of the existing LTI 1.1 implementation, and would barely require any changes in that app.
+
+Instead, we would add an :code:`lti` package to the :code:`openedx.core.content_libraries` app, which will handle LTI endpoints, and communicate with the LTI consumers. It would be a good idea to use a package like `pylti1.3`_ which can handle the LTI-spec implementation, which saves a lot of development and testing effort, and keeps the platform codebase relevant to openedx.
```

Comment:
This is a "decision record", after all, so we shouldn't use phrases like "it would be a good idea".  We just go right out and say: "We will use pylti1.3".

Reply:
Oh, and add an actual link to the library, which seems to be missing here.

Also add information that the library is active and maintained, being used in projects X and Y (with links to them), and has a usable PyPi package.

Reply:
AND (sorry), also link to the PR that originally meant to introduce this library, to demonstrate that 1) the author has community-awareness (a very good thing), and 2) we should feel a little guilty for not having nurtured his PR forward. :P

Reply:
Excellent notes, thanks! I did link the library below btw, you can visit the rendered file at https://github.com/open-craft/edx-platform/blob/mavidser/se-4215-content-lib-lti-adr/openedx/core/djangoapps/content_libraries/docs/decisions/0002-libraries-lti-provider.rst